src/codec.rs

```diff
@@ -1498,4 +1595,14 @@ mod tests {
 		assert_eq!(MyWrapper(3u32.into()).encode(), result);
 		assert_eq!(MyWrapper::decode(&mut &*result).unwrap(), MyWrapper(3_u32.into()));
 	}
+
+	#[test]
+	fn malleability() {
+		let enc = ((5u16 << 2) | 0b01).encode();
+		assert!(Compact::<u16>::decode(&mut & enc[..]).is_err());
```

Comment:
Please check for the exact error message and not just if an error occurred.

And it would be nice to test the other `Compact` impls as well.